BigQuery/tests/System/LoadDataAndQueryTest.php

```diff
@@ -884,6 +884,47 @@ public function testInsertValidJsonToTable($json)
         $this->assertTrue($insertResponse->isSuccessful());
     }
 
+    public function testLoadWithSession()
+    {
+        $dataset = self::$client->dataset('_SESSION');
+        $table = $dataset->table(uniqid(self::TESTING_PREFIX));
+        $source = __DIR__ . '/data/table-data.json';
+        $schema = [
+            'fields' => [
+                [
+                    'name' => 'name',
+                    'type' => 'STRING'
+                ],
+                [
+                    'name' => 'post_abbr',
+                    'type' => 'STRING'
+                ]
+            ]
+        ];
+        $loadJobConfig = $table->load(fopen($source, 'r'))
+            ->sourceFormat('NEWLINE_DELIMITED_JSON')
+            ->autodetect(true)
+            ->createSession(true);
+
+        $job = self::$client->runJob($loadJobConfig);
+
+        $this->assertTrue(isset($job->info()['statistics']['sessionInfo']['sessionId']));
+
+        $sessionId = $job->info()['statistics']['sessionInfo']['sessionId'];
+        $loadJobConfig = $table->load(fopen($source, 'r'))
+            ->sourceFormat('NEWLINE_DELIMITED_JSON')
+            ->connectionProperties(
+                ['key' => 'session_id', 'value' =>$sessionId]
+            );
+        $job = self::$client->runJob($loadJobConfig);
```

Comment:
Can you instead add the `connectionProperties` to the above `runJob` method and assert over its output rather than calling it twice?

Reply:
Actually, we can not provide createSession and connectionProperties in on the same load job. So, I created two jobs to do this. The first one create and gives me a sessionId, second one uses it to do the second load operation to the same temporary table.